controller/QtiCreator.php

```diff
@@ -348,4 +363,66 @@ private function validateXmlInput(string $xml)
         }
         \tao_helpers_Xml::getSimpleXml($xml);
     }
+
+    /**
+     * Download sample template for tabular import
+     *
+     * @param string
+     * @throws common_exception_Error
+     */
+    public function downloadCsv()
+    {
+        if (!$this->hasRequestParameter('uri')) {
+            throw new common_exception_MissingParameter('uri', __METHOD__);
+        }
+
+        $metaDataArray = $this->getMetadataRepository()->getMetadataArray($this->getRequestParameter('uri'));
+        
+        $template = $this->getTemplateRepository()->findById(CsvTemplateRepository::DEFAULT);
+
+        $headers  = $this->getTemplateHeaderParser()->parse($template);
+        $finalHeader = array_merge($headers, $metaDataArray);
+
+        $finalSampleCSV  = $this->getTemplateDataParser()->parser($finalHeader);
+        $filename = $this->getFileName();
```

Comment:
Maybe since we need to manipulate headers, file name based on template, etc we could:

- Use the PSR Response object for it (already available on tao).
- Create a separate class for this

Wdyt?

Reply:
Done